### DISCUSSION TITLE
关于tabs标签v2.2.0后tab-click如何获取绑定的数据

### DISCUSSION THREAD

**Discussion by leechaojie** [element-plus/element-plus#8042] [Q&A] [answered]
v2.2.0之前的版本 tab-click  可以通过 TabsPaneContext 内的 `instance` 找到 tab 上的所有数据，现在我看源码去掉了 `instance`，怎么才能获取 tab 上的所有数据呢？通过 `props` 只能获取 `label` `name`
<img width="877" alt="image" src="https://user-images.githubusercontent.com/28593271/171534661-e954796f-ff61-4cb8-a3fb-e838167f370c.png">

**Comment by dreambo8563** [comment, marked as answer]
没理解这个场景, 你想要的数据应该都是 自己在业务侧已知的吧, 所以没必要从组件里去获取吧?

**Reply by leechaojie** [reply]
标签是动态创建的，每次创建都是根据创建它的逻辑生成一些数据，点击标签的时候要拿到这些数据，

**Reply by dreambo8563** [reply]
是说 panel 里都是不同的组件当做 slot, 点击的时候要拿到 那个组件里的数据? 是吗

**Reply by leechaojie** [reply]
是的，类似于这样的业务逻辑

**Reply by dreambo8563** [reply]
我觉得可以尝试一个思路.
这种我觉得一般都是类似这样的代码, 通过点击后, 通过 contentRef.getxxData(); 来拿到对应数据.  作为 slot 的那个组件. 对外 expose 对应的方法就行.

也可以尝试通过click 的参数 `tab: TabsPaneContext` 来获取到 slot 的引用, 应该也能调用 expose 的方法.
 


```vue
  <el-tab-pane
      v-for="item in editableTabs"
      :key="item.name"
      :label="item.title"
      :name="item.name"
    >
       <component
                ref="contentRef"
                :is="Component"
              />
    </el-tab-pane>
```

**Reply by leechaojie** [reply]
谢谢你，实现了

**Reply by dreambo8563** [reply]
把这个标为 `答案`吧  :)
